cli/src/main/java/com/devonfw/tools/ide/url/model/UrlMetadata.java

```diff
@@ -52,6 +55,32 @@ public UrlEdition getEdition(String tool, String edition) {
     return urlEdition;
   }
 
+  /**
+   * @param tool the name of the {@link UrlTool}.
+   * @return the sorted {@link List} of {@link String editions} .
+   */
+  public List<String> getSortedEditions(String tool) {
+
+    List<String> list = new ArrayList<>();
+    Path path;
+    try {
+      path = this.repository.getChild(tool).getPath();
+    } catch (NullPointerException e) {
+      this.context.warning("Can't get sorted editions for tool {} because it does not exist in {}.", tool,
+          this.repository.getPath());
+      return List.of();
+    }
+    if (Files.isDirectory(path)) {
+      try (Stream<Path> childStream = Files.list(path)) {
+        childStream.forEach(c -> list.add(c.getFileName().toString()));
+      } catch (IOException e) {
+        throw new IllegalStateException("Failed to list children of directory " + path, e);
+      }
+    }
```

Comment:
Why implementing this again instead of reusing `this.repository.getChild(tool).getChildren()` that already provides the list of `UrlEdition` objects that are cached for subsequent calls? As you other where request such edition it will be loaded by `UrlTool` anyhow so the cache prevents multiple directory listings.

Reply:
True, done!